src/preferences/dialog/dlgprefcolors.cpp

```diff
@@ -94,7 +118,84 @@ void DlgPrefColors::slotApply() {
                         m_colorPaletteSettings.getTrackColorPalette()));
     }
 
-    m_pConfig->setValue(
-            ConfigKey("[Controls]", "auto_hotcue_colors"),
-            checkBoxAssignHotcueColors->isChecked());
+    int index = comboBoxHotcueDefaultColor->currentIndex();
+
+    if (index > 0) {
+        m_pConfig->setValue(ConfigKey("[Controls]", "auto_hotcue_colors"), false);
+        m_pConfig->setValue(ConfigKey("[Controls]", "HotcueDefaultColorIndex"), index - 1);
+    } else {
+        m_pConfig->setValue(ConfigKey("[Controls]", "auto_hotcue_colors"), true);
+        m_pConfig->setValue(ConfigKey("[Controls]", "HotcueDefaultColorIndex"), -1);
+    }
+}
+
+QPixmap DlgPrefColors::drawPalettePreview(const QString& paletteName) {
+    foreach (const ColorPalette& palette, mixxx::PredefinedColorPalettes::kPalettes) {
+        if (paletteName == palette.getName()) {
```

Comment:
This limits the drawing of previews to the predefined palettes. I think it would be better to move this function from DlgPrefColors to ColorPalette.

Reply:
I would only do that if we need that preview somewhere else. Right now we don't. But we should use ColotPaletteSettings.getPalette() instead of iterating over the predefined palettes here.